src/main/java/profile/view/SplashView.java

```diff
@@ -0,0 +1,173 @@
+                this,
+                message,
+                "Error",
+                JOptionPane.ERROR_MESSAGE
+        );
+    }
+    
+    @Override
+    public void showSignUpForm() {
+        if (onCreateNewProfile != null) {
+            onCreateNewProfile.run();
+        }
+    }
+    
+    @Override
+    public void setOnCreateNewProfile(Runnable callback) {
+        this.onCreateNewProfile = callback;
+    }
+    
+    private void initializeComponents() {
+        // Create list model and list
+        listModel = new DefaultListModel<>();
+        profileList = new JList<>(listModel);
+        profileList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        profileList.setCellRenderer(new ProfileListCellRenderer());
+        
+        // Create buttons
+        selectButton = new JButton("Select Profile");
+        selectButton.setEnabled(false);
+        
+        createNewButton = new JButton("Create New Profile");
+    }
+    
+    private void setupLayout() {
+        setLayout(new BorderLayout(10, 10));
+        setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
+        
+        // Title
+        JLabel titleLabel = new JLabel("Select User Profile");
+        titleLabel.setHorizontalAlignment(SwingConstants.CENTER);
+        add(titleLabel, BorderLayout.NORTH);
+        
+        // Profile list in scroll pane
+        JScrollPane scrollPane = new JScrollPane(profileList);
+        scrollPane.setPreferredSize(new Dimension(400, 300));
+        scrollPane.setBorder(BorderFactory.createTitledBorder("Available Profiles"));
+        add(scrollPane, BorderLayout.CENTER);
+        
+        // Buttons panel
+        JPanel buttonPanel = new JPanel(new FlowLayout());
+        buttonPanel.add(selectButton);
+        buttonPanel.add(createNewButton);
+        add(buttonPanel, BorderLayout.SOUTH);
+    }
+    
+    private void setupEventHandlers() {
+        // enable/disable select button based on selection
+        profileList.addListSelectionListener(e -> {
+            if (!e.getValueIsAdjusting()) {
+                selectButton.setEnabled(profileList.getSelectedValue() != null);
+            }
+        });
+        
+        // Handle profile selection
+        selectButton.addActionListener(e -> {
+            Profile selectedProfile = profileList.getSelectedValue();
+            if (selectedProfile != null && onProfileSelected != null) {
+                onProfileSelected.accept(selectedProfile);
+            }
+        });
+        
+        
+        // Handle create new profile
+        createNewButton.addActionListener(e -> {
+            if (onCreateNewProfile != null) {
+                onCreateNewProfile.run();
+            }
+        });
+    }
+    
+    /**
+     * show name and basic info.
+     */
+    private static class ProfileListCellRenderer extends DefaultListCellRenderer {
+        @Override
+        public Component getListCellRendererComponent(JList<?> list, Object value,
```

Comment:
Can we change this to specify the type and get rid of the `insanceof` check?

```suggestion
        public Component getListCellRendererComponent(JList<Profile> list, Object value,
```